### PR TITLE
dynamically disable the pay button to prevent amount exceeding the max balance

### PR DESCRIPTION
**Pull request by emmac3**
Using 2 attributes, disable/enable the pay button dynamically without violating existing behaviour

https://user-images.githubusercontent.com/79542071/122605426-a9c3f280-d034-11eb-9272-6426b05bbd50.mov

### CODE REVIEW

**Review by barakj — COMMENTED** [review]
This code doesn't work for the following case:
If I don't select any card on file, and start inputing numbers in the custom text field until I exceed the maximum balance allowed, then start erasing digits until its allowed, the button will now be enabled even though I haven't selected a card to pay with. Pressing on the button would result in a 400 bad request.

**Comment by emmac3** [comment]
> This code doesn't work for the following case:
True.. I'll have to think about this more

File: connect-examples/v2/node_gift-cards/public/js/amount-bar.js [hunk 1]
```diff
@@ -87,8 +87,20 @@ class AmountBar {
...
+
+      payButton.setAttribute("disabled-by-amount", "true");
+    } else {
+      payButton.innerHTML = "Pay " + this.formatMoneyFunction(amount, this.currency, false);
+      document.getElementById("amount-bar__value").value = amount;
+
+      payButton.setAttribute("disabled-by-amount", "false");
+    }
+
+    // HACK: control button in combination with logic from the select dropdown
```
Comment (barakj) [inline connect-examples/v2/node_gift-cards/public/js/amount-bar.js:102]: Let's remove the word hack

File: connect-examples/v2/node_gift-cards/public/js/pretty-select-dropdown.js [hunk 2]
```diff
@@ -248,13 +252,22 @@ class PrettySelectDropdown {
...
 
     valueInput.addEventListener('input', function () {
       if (valueInput.value !== "") {
-        submitButton.disabled = false;
+        submitButton.setAttribute("disabled-by-dropdown", "false");
+      } else {
+        submitButton.setAttribute("disabled-by-dropdown", "true");
+      }
+
+      // HACK: control button in combination with logic from the amount bar
```
Comment (barakj) [inline connect-examples/v2/node_gift-cards/public/js/pretty-select-dropdown.js:264]: likewise

**Review by barakj — COMMENTED** [review]
LGTM but lets clean up the comments per my review

File: connect-examples/v2/node_gift-cards/public/js/amount-bar.js (hunk 1 above)
Comment (emmac3) [reply connect-examples/v2/node_gift-cards/public/js/amount-bar.js:102]: what can we replace it with to indicate that this code is only here for a specific use case?

Comment (barakj) [reply connect-examples/v2/node_gift-cards/public/js/amount-bar.js:102]: How about:
`NOTE: This functionality is used specifically for coordinating logic between the dropdown and the pay button`

Comment (barakj) [reply connect-examples/v2/node_gift-cards/public/js/amount-bar.js:102]: I just think it's better than explicitly using the word hack

Comment (emmac3) [reply connect-examples/v2/node_gift-cards/public/js/amount-bar.js:102]: You are the best!

Comment (barakj) [reply connect-examples/v2/node_gift-cards/public/js/amount-bar.js]: I didn't do anything, you did it all!

**Review by barakj — APPROVED** [review]
LGTM

**Pull request merged**